internal/action/command.go

```diff
@@ -646,6 +647,7 @@ func (h *BufPane) ResetCmd(args []string) {
 	}
 	if _, ok := defaultLocals[option]; ok {
```

Comment:
> It seems to me it is the proper place for that, but above we should probably call `DoSetOptionNative()` instead of `SetOptionNative()`?

We can do so to prevent checking the options validity twice and to do not set `b.LocalSettings` without the need, but then we (currently) set and process it even in the case it hasn't changed for the buffer. At least, if we don't move the equality check to `DoSetOptionNative()`.
It's to be discussed if it's necessary to process the locals without a change.

> It seems this entire `if _, ok := defaultLocals[option]` is superfluous (since `defaultGlobals` includes default locals as well, and `SetGlobalOptionNative()` above sets those default locals per buffer as well)?

Yep, that's right and brings me again to the fact that there is no difference between `DefaultGlobalSettings()` and `DefaultAllSettings()`. Both return exactly the same values, while `DefaultAllSettings()` really should (at least from my perspective) and `DefaultGlobalSettings()` should return only the global-only instead, but then is superfluous since `DefaultGlobalOnlySettings` is already exported?
For the sake of comprehensibility I'd suggest to use `DefaultAllSettings()` within `ResetCmd()` and iterate only over this, as you suggested.

Reply:
> We can do so to prevent checking the options validity twice and to do not set `b.LocalSettings` without the need, but then we (currently) set and process it even in the case it hasn't changed for the buffer. At least, if we don't move the equality check to `DoSetOptionNative()`.
> It's to be discussed if it's necessary to process the locals without a change.

Good point. Hmm, any reason why shouldn't we move the equality check from `SetOptionNative()` to `DoSetOptionNative()` (AND, consistently, from `SetGlobalOptionNative()` to `doSetGlobalOptionNative()`)?



> there is no difference between `DefaultGlobalSettings()` and `DefaultAllSettings()`. Both return exactly the same values

Indeed. Another WTF. (And even bigger WTF is the documentation of all 3 functions: `DefaultCommonSettings()`, `DefaultGlobalSettings()` and `DefaultAllSettings()`.)



> For the sake of comprehensibility I'd suggest to use `DefaultAllSettings()`

I'd be fine with either `DefaultAllSettings()` or `DefaultGlobalSettings()`, as long as its documentation clearly and correctly describes what exactly does it return.

And while we're at it, it seems not hard to clean this up right away, removing either of these two duplicating functions?